packages/blade/src/hooks/useId.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import * as React from 'react';
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect';
+
+/**
+ * Credit: https://github.com/reach/reach-ui/blob/dev/packages/auto-id/src/auto-id.ts
```

Comment:
Didn't used the useId from reach, this version of useId is more simplified since we currently don't support React18. 
If we migrate to React18 we can pull in the NPM package.